### PR TITLE
Snowflake support

### PR DESCRIPTION
**Pull request by ikudjoi**
Support for Snowflake database
https://snowflake.com

I already had placed a PR but I rebased my work on current develop branch, deleted my branch on Github and pushed the new one. But my PR also disappeared. That was probably something that I shouldn't have done, sorry.

I updated the precompiled Snowflake .NET connector binaries and added references to the CLI projects, otherwise this is the same as before, but rebased.

My pull requests to the .NET connector project might get accepted today, so I might be able to replace the references to my own builds with NuGet package sometime soon.

### CODE REVIEW

**Comment by ikudjoi** [comment]
The old PR seems to be still there but was automatically closed.
https://github.com/fluentmigrator/fluentmigrator/pull/944

**Comment by fubar-coder** [comment]
OK, I'll wait a little bit when there is a chance to use a NuGet package for an updated Snowflake connector.

**Comment by ikudjoi** [comment]
Pushed fixes for alter column syntax & more tests. @fubar-coder, have you noticed my private messages in gitter.im?

**Comment by fubar-coder** [comment]
@ikudjoi I saw them just now. Thank you 👍 .

**Comment by ikudjoi** [comment]
I had to do a lot of changes to the test because I thought there is no default schema in Snowflake. But today I learned I was wrong, there is one. New version supporting default schema is now pushed.

**Comment by ikudjoi** [comment]
And now the new version of the .NET connector is out, removed own builds and replaced them with the new NuGet package.

**Comment by ikudjoi** [comment]
I'm not sure if you want to embrace the latest commit I just pushed. I ran into a problem when using the FM without quoting identifiers. Snowflake converts unquoted identifiers to uppercase, so the versioninfo table contains column "VERSION". But FluentMigrator is trying to read column "Version".

I suggest it would not be too harmful to match the column case insensitive in this context. I found no other way to match it case insensitive, setting caseSensitive = false didn't work.

File: src/FluentMigrator.Runner/VersionLoader.cs [hunk 1]
```diff
@@ -147,9 +147,23 @@ public void LoadVersionInfo()
             if (!AlreadyCreatedVersionTable) return;
 
             var dataSet = _processor.ReadTableData(VersionTableMetaData.SchemaName, VersionTableMetaData.TableName);
-            foreach (DataRow row in dataSet.Tables[0].Rows)
+            var dataTable = dataSet.Tables[0];
+
+            // Ignore case when looking for correct column.
+            foreach (DataColumn col in dataTable.Columns)
```
Comment (fubar-coder) [inline src/FluentMigrator.Runner/VersionLoader.cs]: This change is not optimal. It might be easier to just set `dataTable.CaseSensitive` to `true`

Comment (fubar-coder) [reply src/FluentMigrator.Runner/VersionLoader.cs]: > setting caseSensitive = false didn't work.

Uh, ok. But it still looks strange. Just get the column index and use that instead - but only if `Version` couldn't be found.

Comment (fubar-coder) [reply src/FluentMigrator.Runner/VersionLoader.cs]: And please use `OrdinalIgnoreCase`.

Comment (ikudjoi) [reply src/FluentMigrator.Runner/VersionLoader.cs]: I'm trying to find better solution.

File: src/FluentMigrator.Runner/VersionLoader.cs [hunk 2]
```diff
@@ -147,9 +147,28 @@ public void LoadVersionInfo()
             if (!AlreadyCreatedVersionTable) return;
 
             var dataSet = _processor.ReadTableData(VersionTableMetaData.SchemaName, VersionTableMetaData.TableName);
-            foreach (DataRow row in dataSet.Tables[0].Rows)
+            var dataTable = dataSet.Tables[0];
+            var versionColumnIndex = dataTable.Columns.IndexOf(VersionTableMetaData.ColumnName);
+
+            // Find out correct column by case insensitive matching if column was not found. Setting dataTable.caseSensitive = false does not help for some reason.
+            if (versionColumnIndex < 0)
```
Comment (fubar-coder) [inline src/FluentMigrator.Runner/VersionLoader.cs]: Reminder for me: Check for -1 instead.

File: test/FluentMigrator.Tests/FluentMigrator.Tests.csproj [hunk 3]
```diff
@@ -76,6 +78,10 @@
       <Link>%(Filename)%(Extension)</Link>
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </Content>
+    <Content Include="../../lib/Snowflake/*.*">
```
Comment (fubar-coder) [inline test/FluentMigrator.Tests/FluentMigrator.Tests.csproj]: Reminder for me: Check if it's still required

Comment (ikudjoi) [reply test/FluentMigrator.Tests/FluentMigrator.Tests.csproj]: That folder was removed, so no, it is not. Thanks for accepting the PR!

**Pull request merged**